### PR TITLE
Reproducible builds refactor

### PR DESCRIPTION
**Pull request by hanneary**
# Why
- Refactor reproducible builds to use build kit instead of kaniko, buildkit is much faster and better supported
- Use manipulated enclave.Dockerfile (formally known as ev-user.Dockerfile ) as the source of truth versioning instead of tracking each version in the cage.toml

# How
- Use the buildkit in docker - this means the user needs to have a docker version that supports buildx v0.10.0 which corresponds to buildkit 11. Cages can still be built with lower versions but theres a warning that they won't be reproducible.
- Came full circle from using the buildctl daemon because it requires sudo for setup, once buildx is over v0.10.0 builds should be repro
- There are some work arounds around file dates and docker whiteouts you'll see at the end of [testRepro.Dockerfile](https://github.com/evervault/cage-cli/compare/hannah/spe-531-try-swap-kaniko-for-buildkit?expand=1#diff-af8bb800bbe12409cc2bae81bb1d1b7c266647a7f02bf5492b3401339fd773f1) full explanation [here](https://medium.com/nttlabs/bit-for-bit-reproducible-builds-with-dockerfile-7cc2b9faed9f)

### CODE REVIEW

File: .DS_Store
Comment (donaltuohy) [inline .DS_Store:1]: We should put this file in gitignore

**Review by donaltuohy — APPROVED** [review]
Nice 🔥

File: .DS_Store
Comment (hanneary) [reply .DS_Store:1]: Oh yes will do now!

Comment (ghost) [reply .DS_Store:1]: Might be an idea to add https://www.toptal.com/developers/gitignore/api/macos,visualstudiocode,rust

Comment (hanneary) [reply .DS_Store:1]: good shout added there

**Pull request merged**